### PR TITLE
[MRG] Improve readability of changelog with github subsitution

### PR DESCRIPTION
**Pull request by jasmainak**

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -18,9 +18,10 @@ Simulation (:py:mod:`hnn_core`):
    Network
    Cell
    CellResponse
+   pick_connection
 
-Published Models (:py:mod:`hnn_core`):
---------------------------------------
+Network Models (:py:mod:`hnn_core`):
```
Comment (jasmainak) [inline doc/api.rst:23]: @ntolley I renamed this so there were not too many categories. Let me know if this is okay for you

**Comment by jasmainak** [comment]
@ntolley @rythorpe @cjayb I think it would be nice to summarize a list of notable changes like we do in `mne-bids`:

https://mne.tools/mne-bids/stable/whats_new_previous_releases.html#id7

before the release so it's more digestible for the users. If one of you wants to take the lead on this, please feel free!

**Comment by jasmainak** [comment]
any reviews/opinions here or should we merge? I just copied this from `mne-bids`. Basically it helps avoid incorrect links in our whats_new.rst

**Comment by cjayb** [comment]
Sweet sugar, thanks @jasmainak :)

**Pull request merged**